buckets-mdapi/src/gc.rs

```diff
@@ -1,8 +1,35 @@
-// Copyright 2019 Joyent, Inc.
+// Copyright 2020 Joyent, Inc.
+
+use uuid::Uuid;
+
+use crate::types::RowSlice;
 
 pub mod delete;
 pub mod get;
 
-pub fn refresh_garbage_view_sql() -> &'static str {
+pub(crate) fn refresh_garbage_view_sql() -> &'static str {
     "REFRESH MATERIALIZED VIEW GARBAGE_BATCH"
 }
+
+pub(crate) fn get_garbage_batch_id_sql() -> &'static str {
+    "SELECT batch_id FROM garbage_batch_id WHERE id = 1"
+}
+
+pub(crate) fn update_garbage_batch_id_sql() -> &'static str {
+    "UPDATE garbage_batch_id SET batch_id = $1 WHERE id = 1"
+}
+
+pub(crate) fn handle_batch_id_result(rows: &RowSlice) -> Result<Uuid, String> {
+    if rows.len() == 1 {
+        // let mut batch_id: Option<Uuid> = None;
+        let batch_id = rows[0].get("batch_id");
+        Ok(batch_id)
+    } else if rows.is_empty() {
+        Err("garbage batch id not found".into())
+    } else {
+        Err(
+            "database invariant failure: found more than one garbage batch id"
```

Comment:
Curious what happens in this case? Does it need manual cleanup?

Reply:
Yeah, that would require some manual intervention. Basically it would mean that we've gotten multiple records that match a query done on the table's primary key. That ought to never happen, but we still handle it for completeness. I'll also add a comment here to explain it a bit.

Reply:
Oops, I take that back. There isn't a primary key established for the `garbage_batch_id` table though there probably should be. I'll go ahead and add a new migration to add that constraint as part of this change.

Reply:
Migration and comment added